kie-remote/kie-remote-client/src/main/java/org/kie/services/client/api/command/AbstractRemoteCommandObject.java

```diff
@@ -228,15 +228,19 @@ void preprocessParameterCommand( Object cmdObj, List<Object> extraClassInstanceL
 
     void addPossiblyNullObject( Object inputObject, List<Object> objectList ) {
         if( inputObject != null ) {
-            if( inputObject instanceof List )  { 
-                objectList.addAll((List) inputObject);
+            if( inputObject instanceof List )  {
+                for( Object obj : (List<?>)inputObject ) {
+                    addPossiblyNullObject(obj, objectList);
+                }
```

Comment:
Could you add a check to see if the object isn't already present in `objectList` -- users sometimes do crazy things, including trying to submit self-referential nested lists, which would lead to a stack-overflow in this case. :/ (If the object has already been added, an exception should probably be thrown). 

Good catch, Karel!